snapcraft/internal/meta.py

```diff
@@ -458,3 +461,14 @@ def _validate_hook(hook_path):
     if not os.stat(hook_path).st_mode & stat.S_IEXEC:
         asset = os.path.basename(hook_path)
         raise CommandError('hook {!r} is not executable'.format(asset))
+
+
+def _verify_app_paths(basedir, apps):
+    for app in apps:
+        path_entries = [i for i in ('desktop', 'completer')
```

Comment:
Right. What I mean is, can we use the values from the schema? Since this code looks like something that's very to track down and nothing says what it means and how these entries are meaningful.